test/intest7.F90

```diff
@@ -0,0 +1,142 @@
+! This is a test for NCEPLIBS-bufr.
+!
+! Reads test file 'testfiles/IN_7' containing 2-03-YYY changed reference values, using inline ERRWRT to
+! check error messages, and using UFBPOS, UFBTAB, and VALX
+!
+! J. Ator, 2/23/2023
+
+module Share_errstr
+  ! This module is needed in order to share information between the test program and subroutine errwrt, because
+  ! the latter is not called by the former but rather is called directly from within the BUFRLIB software.
+
+  character*1500 errstr
+
+  integer errstr_len
+end module Share_errstr
+
+subroutine errwrt(str)
+  ! This subroutine supersedes the subroutine of the same name within the BUFRLIB software, so that we can
+  ! easily test the generation of error messages from within the library.
+
+  use Share_errstr
+
+  character*(*) str
+
+  integer str_len
+
+  str_len = len(str)
+  errstr ( errstr_len + 1 : errstr_len + str_len + 1 ) = str
+  errstr_len = errstr_len + str_len
+
+  return
+end subroutine errwrt
+
+program intest7
+  use Share_errstr
+
+  implicit none
+
+  integer*4 isetprm, igetprm, ireadns, ibfms
+
+  integer imgdt, iret1, iret2, jdate, nr8a, nr8v, nr8v2, idx1, idx2, nsub
+
+  integer mxr8pm, mxr8lv
+  parameter ( mxr8pm = 15 )
+  parameter ( mxr8lv = 5 )
+
+  real*8 r8arr (mxr8pm, mxr8lv), r8val
+
+  real valx
+
+  character cmgtag*8
+
+  print *, 'Testing reading IN_7 containing 2-03-YYY changed reference values, using inline ERRWRT'
+  print *, 'to check error messages, and using UFBPOS, UFBTAB, and VALX'
+
+#ifdef KIND_8
+  call setim8b ( .true. )
+#endif
+
+  ! Check error messages in ISETPRM.
+  iret1 = isetprm ( 'MXNRV', 5 )
+  errstr_len = 1
+  iret2 = isetprm ( 'DUMMY', 20 )
+  if ( ( iret1 .ne. 0 ) .or. ( iret2 .ne. -1 ) .or. &
+      ( index( errstr(1:errstr_len), 'ISETPRM - UNKNOWN INPUT PARAMETER DUMMY' ) .eq. 0 ) ) stop 1
+
+  ! Open the input file and DX table.
+  open ( unit = 11, file = 'testfiles/IN_7', form ='unformatted')
+  open ( unit = 12, file = 'testfiles/IN_7_bufrtab' )
+  call openbf ( 11, 'IN', 12 )
+  call openbf ( 11, 'QUIET', 1 )
+
+  ! Check error messages in IGETPRM.
+  iret1 = igetprm ( 'MXNRV' )
+  errstr_len = 1
+  iret2 = igetprm ( 'DUMMY' )
+  if ( ( iret1 .ne. 5 ) .or. ( iret2 .ne. -1 ) .or. &
```

Comment:
Check results as soon as you get them. So test iret1 right after linke 74 instead of waiting to call igetprm() again.

It should be generally the case that after every call to the library in a test, you should test every value that can be tested. before calling another library function.

Reply:
OK, good suggestion, and will do